process/block/sovereignChainBlock.go

```diff
@@ -848,6 +869,14 @@ func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() er
 	return nil
 }
 
+// this will return true if we receive the genesis main chain header in following cases:
+//   - no notifier is attached => we did not track main chain and don't have pre-genesis header
+//   - node is in re-sync/start in the exact epoch when we start to notarize main chain => no previous
+//     main chain tracking(notifier is also disabled)
+func (scbp *sovereignChainBlockProcessor) receivedGenesisMainChainHeaderWithoutPreGenesis(incomingHeader data.HeaderHandler) bool {
```

Comment:
this function name is really confusing. Can you find a better name?

Reply:
Right, good point, changed it